paddle/platform/device_context_manager.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#pragma once
+#include "paddle/platform/device_context.h"
+
+namespace paddle {
+namespace platform {
+
+template <typename T>
+struct Converter;
+
+template <>
+struct Converter<CPUPlace> {
+  using DeviceContextType = CPUDeviceContext;
+};
+
+#ifndef PADDLE_ONLY_CPU
+template <>
+struct Converter<GPUPlace> {
+  using DeviceContextType = CUDADeviceContext;
+};
+#endif
+
+class DeviceContextManager {
```

Comment:
DeviceContextManager implements a singleton container of pre-created device contexts. Let us don't introduce such a global variable. Instead, let us create device context from Place in Executor.